qutebrowser/mainwindow/mainwindow.py

```diff
@@ -493,6 +497,15 @@ def _connect_signals(self):
             completion_obj.on_clear_completion_selection)
         cmd.hide_completion.connect(completion_obj.hide)
 
+    def _set_decoration(self, hidden):
+        """ Set the visibility of the window decoration via Qt."""
+        window_flags = Qt.Window
+        if hidden:
+            window_flags |= Qt.CustomizeWindowHint | Qt.NoDropShadowWindowHint
+        self.setWindowFlags(window_flags)
+        self.hide()
+        self.show()
```

Comment:
I think the window now always shows up when creating it because of this - probably, `self.hide()`/`self.show()` should be guarded by a `if self.isVisible():`?

Reply:
After `_on_config_changed` is called on the window, it seems like the window is hidden. At least, `self.isVisible()` always returns `False`, both on startup and when changing the config option via the GUI. I'll try to investigate this myself a little bit further though.